crates/matrix-sdk/src/widget/machine/mod.rs

```diff
@@ -314,83 +353,79 @@ impl WidgetMachine {
                 }
-
-                const DEFAULT_EVENT_LIMIT: u32 = 50;
-                let limit = limit.unwrap_or(DEFAULT_EVENT_LIMIT);
-                let request = ReadMessageLikeEventRequest { event_type, limit };
-
-                let (request, action) = self.send_matrix_driver_request(request);
-
-                request.then(|result, machine| {
-                    let response = match &machine.capabilities {
-                        CapabilitiesState::Unset => Err(FromWidgetErrorResponse::from_string(
-                            "Received read event request before capabilities negotiation",
-                        )),
-                        CapabilitiesState::Negotiating => {
-                            Err(FromWidgetErrorResponse::from_string(
-                                "Received read event request while capabilities were negotiating",
-                            ))
-                        }
-                        CapabilitiesState::Negotiated(capabilities) => result
-                            .map(|mut events| {
-                                events.retain(|e| capabilities.raw_event_matches_read_filter(e));
-                                ReadEventResponse { events }
-                            })
-                            .map_err(FromWidgetErrorResponse::from_error),
-                    };
-
-                    vec![Self::send_from_widget_response(raw_request, response)]
-                });
-
-                action
             }
-
-            ReadEventRequest::ReadStateEvent { event_type, state_key } => {
+            Some(state_key) => {
                 let allowed = match &state_key {
-                    StateKeySelector::Any => capabilities
-                        .read
-                        .iter()
-                        .any(|filter| filter.matches_state_event_with_any_state_key(&event_type)),
+                    StateKeySelector::Any => capabilities.read.iter().any(|filter| {
+                        filter.matches_state_event_with_any_state_key(
+                            &request.event_type.to_string().into(),
+                        )
+                    }),
 
                     StateKeySelector::Key(state_key) => {
                         let filter_in = MatrixEventFilterInput {
-                            event_type: event_type.to_string().into(),
+                            event_type: request.event_type.clone(),
                             state_key: Some(state_key.clone()),
                             // content doesn't matter for state events
                             content: MatrixEventContent::default(),
                         };
 
                         capabilities.read.iter().any(|filter| filter.matches(&filter_in))
                     }
-                };
+                } || capabilities.read.iter().any(|f| {
+                    f.matches_message_like_event_type(&request.event_type.to_string().into())
+                });
```

Comment:
Can you use a local variable, or break the if into two ifs? using a match statement as a bool expression in a compound condition makes the code harder to read.